vertical-pod-autoscaler/pkg/recommender/input/cluster_feeder_test.go

```diff
@@ -89,22 +119,92 @@ func TestLegacySelector(t *testing.T) {
 			expectedConfigUnsupported: nil,
 			expectedConfigDeprecated:  nil,
 		}, {
+			name:                      "new and legacy selector",
 			legacySelector:            parseLabelSelector("app = test1"),
 			selector:                  parseLabelSelector("app = test2"),
 			fetchSelectorError:        nil,
 			expectedSelector:          labels.Nothing(),
 			expectedConfigUnsupported: &unsupportedConditionBothDefined,
 			expectedConfigDeprecated:  nil,
 		},
+		{
+			name:               "can't decide if top-level-ref",
+			legacySelector:     nil,
+			selector:           nil,
+			fetchSelectorError: nil,
+			expectedSelector:   labels.Nothing(),
+			targetRef: &v1.CrossVersionObjectReference{
+				Kind:       kind,
+				Name:       name1,
+				APIVersion: apiVersion,
+			},
+			expectedConfigUnsupported: &unsupportedConditionNoTargetRef,
+		},
+		{
+			name:               "non-top-level target ref",
+			legacySelector:     nil,
+			selector:           parseLabelSelector("app = test"),
+			fetchSelectorError: nil,
+			expectedSelector:   labels.Nothing(),
+			targetRef: &v1.CrossVersionObjectReference{
+				Kind:       kind,
+				Name:       name1,
+				APIVersion: apiVersion,
+			},
+			topLevelKey: &controllerfetcher.ControllerKeyWithAPIVersion{
+				ControllerKey: controllerfetcher.ControllerKey{
+					Kind:      kind,
+					Name:      name2,
+					Namespace: namespace,
+				},
+				ApiVersion: apiVersion,
+			},
+			expectedConfigUnsupported: &unsupportedTargetRefHasParent,
+		},
+		{
+			name:               "error checking if top-level-ref",
+			legacySelector:     nil,
+			selector:           nil,
+			fetchSelectorError: nil,
+			expectedSelector:   labels.Nothing(),
+			targetRef: &v1.CrossVersionObjectReference{
+				Kind:       "doestar",
+				Name:       "doseph-doestar",
+				APIVersion: "taxonomy",
+			},
+			expectedConfigUnsupported: &unsupportedConditionMudaMudaMuda,
+			findTopLevelError:         fmt.Errorf("muda muda muda"),
+		},
+		{
+			name:               "top-level target ref",
+			legacySelector:     nil,
+			selector:           parseLabelSelector("app = test"),
+			fetchSelectorError: nil,
+			expectedSelector:   parseLabelSelector("app = test"),
+			targetRef: &v1.CrossVersionObjectReference{
+				Kind:       kind,
+				Name:       name1,
+				APIVersion: apiVersion,
+			},
+			topLevelKey: &controllerfetcher.ControllerKeyWithAPIVersion{
+				ControllerKey: controllerfetcher.ControllerKey{
+					Kind:      kind,
+					Name:      name1,
+					Namespace: namespace,
+				},
+				ApiVersion: apiVersion,
+			},
+			expectedConfigUnsupported: nil,
+		},
 	}
 
-	for i, tc := range testCases {
+	for _, tc := range testCases {
 
-		t.Run(fmt.Sprintf("test case number: %d", i), func(t *testing.T) {
+		t.Run(fmt.Sprintf("test case number: %s", tc.name), func(t *testing.T) {
```

Comment:
nit: You can skip the "test case number" i.e. just do t.Run(tc.name, ...

Reply:
Done.